metacat-main/src/main/java/com/netflix/metacat/main/services/search/ElasticSearchUtilImpl.java

```diff
@@ -118,16 +126,53 @@ public ElasticSearchUtilImpl(
     public void delete(final String type, final String id) {
         try {
             RETRY_ES_PUBLISH.call(() -> {
-                client.prepareDelete(esIndex, type, id).execute().actionGet();
+                client.prepareDelete(esIndex, type, id).execute().actionGet(esCallTimeout);
                 return null;
             });
         } catch (Exception e) {
-            log.error("Failed deleting metadata of type {} with id {}. {}", type, id, e);
-            this.elasticSearchMetric.getElasticSearchDeleteFailureCounter().increment();
-            log("ElasticSearchUtil.delete", type, id, null, e.getMessage(), e, true);
+            handleException("ElasticSearchUtil.delete", type, id, e,
+                Metrics.CounterElasticSearchDelete.getMetricName());
         }
     }
 
+    private void handleException(final String request,
+                                 final String type,
+                                 final String id,
+                                 final Exception exception,
+                                 final String metricName) {
+        log.error("Failed {} metadata of type {} with id {}. {}", request, type, id, exception);
+        String exceptionName = exception.getClass().getSimpleName();
+        if (exception instanceof RetryException) {
+            final Throwable error = ((RetryException) exception).getLastFailedAttempt().getExceptionCause();
+            if (error != null) {
+                exceptionName = error.getClass().getSimpleName();
+            }
+        }
+        final Map<String, String> tags = ImmutableMap
+            .<String, String>builder().put("status", "failure").put("name", id).put("exception", exceptionName).build();
+        registry.counter(registry.createId(metricName).withTags(tags)).increment();
+        log(request, type, id, null, exception.getMessage(), exception, true);
+    }
```

Comment:
The only difference between these two functions is the 'id' vs 'ids' parameter and how we use them in the tag and log. My suggestion is to rename the second one to handleExceptionForBulkOperation or something similar.